python/tests/integration/connectors/memory/test_postgres.py

```diff
@@ -0,0 +1,250 @@
+        additional_metadata="additional metadata",
+        external_source_name="external source",
+        timestamp=str(datetime.now()),
+    )
+
+
+@pytest.fixture
+def memory_record3():
+    return MemoryRecord(
+        id="test_id3",
+        text="sample text3",
+        is_reference=False,
+        embedding=np.array([0.25, 0.80]),
+        description="description",
+        additional_metadata="additional metadata",
+        external_source_name="external source",
+        timestamp=str(datetime.now()),
+    )
+
+
+def test_constructor(get_postgres_config):
+    connection_string = get_postgres_config
+    memory = PostgresMemoryStore(connection_string, 1, 5)
+    assert memory._connection_pool is not None
+
+
+@pytest.mark.asyncio
+async def test_create_and_does_collection_exist_async(get_postgres_config):
+    connection_string = get_postgres_config
+    memory = PostgresMemoryStore(connection_string, 1, 5)
+
+    await memory.create_collection_async("test_collection", 2)
+    result = await memory.does_collection_exist_async("test_collection")
+    assert result is not None
+
+
+@pytest.mark.asyncio
+async def test_get_collections_async(get_postgres_config):
+    connection_string = get_postgres_config
+    memory = PostgresMemoryStore(connection_string, 1, 5)
+
+    await memory.create_collection_async("test_collection", 2)
+    # await memory.create_collection_async("test_collection-b")
+    # await memory.create_collection_async("test_collection-c")
+    result = await memory.get_collections_async()
+    assert len(result) > 1
+
+
+@pytest.mark.asyncio
+async def test_delete_collection_async(get_postgres_config):
+    connection_string = get_postgres_config
+    memory = PostgresMemoryStore(connection_string, 1, 5)
+
+    await memory.create_collection_async("test_collection", 2)
+    await memory.delete_collection_async("test_collection")
+    result = await memory.get_collections_async()
+    assert "test_collection" not in result
+
+
+@pytest.mark.asyncio
+async def test_does_collection_exist_async(get_postgres_config):
+    connection_string = get_postgres_config
+    memory = PostgresMemoryStore(connection_string, 1, 5)
+
+    await memory.create_collection_async("test_collection", 2)
+    result = await memory.does_collection_exist_async("test_collection")
+    assert result is True
+
+
+@pytest.mark.asyncio
+async def test_upsert_async_and_get_async(get_postgres_config, memory_record1):
+    connection_string = get_postgres_config
+    memory = PostgresMemoryStore(connection_string, 1, 5)
+
+    await memory.create_collection_async("test_collection", 2)
+    await memory.upsert_async("test_collection", memory_record1)
+    result = await memory.get_async(
+        "test_collection", memory_record1._id, with_embedding=True
+    )
+    assert result is not None
+    assert result._id == memory_record1._id
+    assert result._text == memory_record1._text
+    assert result.embedding is not None
```

Comment:
Should we also verify how embedding vector was stored by checking actual vector values?